core/src/main/java/nav/statusplattform/core/repositories/RecordRepository.java

```diff
@@ -99,6 +95,28 @@ public List<RecordEntity> getRecordHistory(UUID serviceId, int maxNumberOfRecord
                 .list(RecordRepository::toRecord);
     }
 
+    public final List<RecordEntity> getRecordsInTimeSpan(UUID serviceId, ZonedDateTime from, ZonedDateTime to) {
+        System.out.println("getRecordsInTimeSpan");
+
+        Optional<RecordEntity> recordEntity = recordTable.where("service_Id", serviceId)
+                .whereExpression("created_at <= ?", from)
+                .orderBy("created_at DESC")
+                .limit(1)
+                .singleObject(RecordRepository::toRecord);
+
+        List<RecordEntity> recordEntities = new ArrayList<>();
+
+        recordEntity.ifPresent(recordEntities::add);
+
+        recordEntities = recordTable.where("service_Id", serviceId)
+                .whereExpression("created_at <= ?", from)
+                .whereExpression("created_at >= ?", to)
+                .orderBy("created_at ASC")
+                .list(RecordRepository::toRecord);
```

Comment:
The assignment on line 111 removes the `Optionally` added `recordEntity`.

Maybe use more descriptive names, like say `recordActiveAtStart`, `recordsInTimespan` or such.

Reply:
Code ensure the record proceeding the search is returned and is added as the first record in the returning results sort order.

Reply:
In addition, returning record given a recognisable name regarding its content.